bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -100,6 +99,7 @@
     private LedgerMetadata metadata;
     final long ledgerId;
     long lastAddPushed;
+    boolean closing;
```

Comment:
Looks like because we don't update the ephemeral metadata representation until the new state is persistent, we need to remember that we kicked off such a close.  I'm a bit confused as to why asyncReadLastConfirmed only needs to check the LedgerMatadata.isClosed() method.  Seems like it should be uniform -- all client calls fail once the LedgerHandle close process has begun -- in which case probably there should be a helper which wraps the closing || LedgerMatdata.isClosed() check.

Reply:
It might also clarify matters to have a 3 value state (OPEN, CLOSING, CLOSED) reflecting the three possible values of (closing, LedgerMetadata.isClosed()) -- namely (false, false), (true, false), (true, true).

Reply:
let's add a handleState, that's probably making things clearer.